plugins/auth-backend-module-guest-provider/src/module.ts

```diff
@@ -36,6 +36,11 @@ export const authModuleGuestProvider = createBackendModule({
         config: coreServices.rootConfig,
       },
       async init({ providers, logger, config }) {
+        const providerConfig = config.getOptionalConfig('auth.providers.guest');
+        if (!providerConfig) {
+          return;
+        }
```

Comment:
Alright, we talked about this one among the maintainers and we agree that the initial log is just plain wrong. Would you mind to update this PR to do things a bit differently, making it better for everyone? Would appreciate it 🙏 

Summary:

- Remove this added code, and also remove the `if` and logging below. Essentially leave _only_ the `providers.registerProvider` line in here
- Modify [the initialize function of the authenticator](https://github.com/backstage/backstage/blob/e9180e8b757e295cb11325ae0387e07fc1408a33/plugins/auth-backend-module-guest-provider/src/authenticator.ts#L23) instead
  - change it to `initialize({ config })` - it gets that passed in
  - make it check whether it's outside of the development env, AND `dangerouslyAllowOutsideDevelopment` isn't set to true
  - if so, throw a hard error similar to the one below, like `The guest provider cannot be used outside of a development environment`, which rejects the login

Reply:
I do like making thing better for everyone!🤗 I'll update my PR as suggested 😊

Reply:
Updated with your suggestions <3